### DISCUSSION TITLE
Allow players to transfer claims

### DISCUSSION THREAD

**Discussion by DrSnowPengu** [GriefPrevention/GriefPrevention#74] [Ideas]
Grief Prevention Author I have really good suggestion. I was wondering If you can add command where players can give there own claims to other players. Is there way you can add that command? I think it would be great idea. :)

Thank You!

**Comment by bigpresh** [comment]
There's already the transferclaim command which allows admins with that permission to transfer claims.  It should be reasonably easy, I'd think, to make that command executable by everyone, and add a check in it that either the claim being transferred belongs to the person making the transfer, or the person making the transfer has the existing permission to transfer any claim.

HOWEVER: it opens up a griefing potential, by users handing claims to other people that the other person might not want - for instance, a claim containing something offensive that other players would think the new owner was responsible for, or simply transferring claims they don't want and thereby taking up their claim blocks so they can't claim areas they do want until they get all the way to wherever the unwanted claim is and drop it.

So, I imagine it'd have to be either disabled for player-player transfers by default, and only enabled by admins who know the risks and are prepared to accept them, or would need some verification from the "gaining" player first, which would somewhat complicate matters.  (Perhaps it could require both players to be online, and a "losing" player using the command would record in memory the claim ID and transfer details, and message the "gaining" player prompting them to use a command like `/accepttransfer $random_id` to "receive" the claim if they want it, and otherwise to ignore it.

**Comment by Magnum97** [comment]
I have been trying to work out how this could work on my server as well. The problem that I run into currently is that if they do not add a user name it will create an admin claim even without the executing user having the griefprevetnion.adminclaim permission. I would be happy to see transfers allowed if admin claims were not allowed by users without permission. They are not allowed to transfer an admin claim to someone else which is good. 

@bigpresh brings up good points about the griefing potentials of allowing this without the receiving user accepting the transfer. In my situation that is not a big issue but I do see the potential for abuse. 

I am happy to hear that adding checks to see if the transferring player owns the claim would be easy. I would be very happy to see this feature added.
